dspace-api/src/main/java/org/dspace/submit/step/ElsevierUploadStep.java

```diff
@@ -0,0 +1,509 @@
+            if (status != STATUS_COMPLETE)
+            {
+                return status;
+            }
+
+            // remove current bitstream from Submission Info
+            subInfo.setBitstream(null);
+        }
+
+        // -------------------------------------------------
+        // Step #3: Check for a change in file description
+        // -------------------------------------------------
+        String fileDescription = request.getParameter("description");
+
+        if (fileDescription != null && fileDescription.length() > 0)
+        {
+            // save this file description
+            int status = processSaveFileDescription(context, request, response,
+                    subInfo);
+
+            // if error occurred, return immediately
+            if (status != STATUS_COMPLETE)
+            {
+                return status;
+            }
+        }
+
+        // ------------------------------------------
+        // Step #4: Check for a file format change
+        // (if user had to manually specify format)
+        // ------------------------------------------
+        int formatTypeID = Util.getIntParameter(request, "format");
+        String formatDesc = request.getParameter("format_description");
+
+        // if a format id or description was found, then save this format!
+        if (formatTypeID >= 0
+                || (formatDesc != null && formatDesc.length() > 0))
+        {
+            // save this specified format
+            int status = processSaveFileFormat(context, request, response,
+                    subInfo);
+
+            // if error occurred, return immediately
+            if (status != STATUS_COMPLETE)
+            {
+                return status;
+            }
+        }
+
+        if (bitstreamId != null) {
+            String fileAccess = request.getParameter("file-access");
+            if (fileAccess == null) {
+                return STATUS_NO_FIlE_ACCESS_ERROR;
+            } else {
+                Bitstream b = bitstreamService
+                        .find(context, UUID.fromString(bitstreamId));
+                bitstreamService.setMetadataSingleValue(context,b,"workflow","fileaccess",null,null,fileAccess);
+
+                if(fileAccess.equals("embargo")) {
+                    DCDate embargoDate = fileAccessFromMetadataService.getEmbargoDate(request);
+
+                    if (embargoDate.toDate() != null) {
+                        bitstreamService.setMetadataSingleValue(context, b, "workflow","fileaccess","date", null, embargoDate.toString());
+                    }
+
+                    if(fileAccessFromMetadataService.fileAccessIdentical(context,b)){
+                        bitstreamService.setMetadataSingleValue(context, b, "workflow","fileaccess","overruled", null, "false");
+
+                    }
+                    else {
+                        bitstreamService.setMetadataSingleValue(context, b, "workflow", "fileaccess", "overruled", null, "true");
```

Comment:
Doing another code review today...

Why are we tracking whether fileaccess rights are overruled?  Maybe I'm misunderstanding the purpose of the `ElsevierUploadStep`.  Initially, I thought it was just pulling down access rights / embargo info via the Elsevier API. But, it also seems to be automatically tracking things in `workflow.fileaccess.date` (which seems duplicative of the actual embargo info in the Resource Policies), and in `workflow.fileaccess.overruled` (whose purpose doesn't make sense to me). What is the purpose of these new "workflow.fileaccess" metadata fields?

Sorry to only be noticing this so late in the game, but I'm not wondering what *are* the use cases where someone would want to enable the ElsevierUploadStep?  Is it required for everything else to function?